installer/tue-install-impl.bash

```diff
@@ -113,6 +113,85 @@ function tue-install-pipe
+    local parent_target old_dep_target error_code
+    parent_target=$1
+    old_dep_target=$2
+    error_code=0
+
+    # Remove the parent from the dependants of the old dep
+    local old_dep_dep_on_file
+    old_dep_dep_on_file="${TUE_INSTALL_DEPENDENCIES_ON_DIR}"/"${old_dep_target}"
+    if [[ -f "${old_dep_dep_on_file}" ]]
+    then
+        # Target is removed, so remove yourself from depend-on files of deps
+        local tmp_dep_on_file found_target
+        tmp_dep_on_file=/tmp/temp_depend_on
+        found_target=false
+        while read -r line
+        do
+            if [[ ${line} != "${parent_target}" ]]
+            then
+                echo "${line}"
+            else
+                found_target=true
+            fi
+        done <"${old_dep_dep_on_file}" >"${tmp_dep_on_file}"
+
+        if ${found_target}
+        then
+            mv "${tmp_dep_on_file}" "${old_dep_dep_on_file}"
+            tue-install-debug "[remove_old_dep] Removed '${parent_target}' from depend-on file of '${old_dep_target}'"
+        else
+            tue-install-warning "[remove_old_dep] '${parent_target}' depended on '${old_dep_target}', so ${old_dep_dep_on_file} should existed with '${parent_target}' in it"
+        fi
+    else
+        tue-install-warning "[remove_old_dep] No dependencies-on file exist for old dependency '${old_dep_target}' of ${parent_target}'"
+    fi
+
+    # When the old dep has no other dependants anymore, remove it, including it deps
+    if [[ -f "${old_dep_dep_on_file}" ]] && [[ -n $(cat "${old_dep_dep_on_file}") ]]
+    then
+        # depend-on is not empty, so not doing anything
+        tue-install-debug "[remove_old_dep] Other targets still depend on ${old_dep_target}, so ignoring it"
+        return 0
+    else
+        # depend-on is empty, so remove it and continue to actual removing of the target
+        tue-install-debug "[remove_old_dep] Deleting empty depend-on file of: ${old_dep_target}"
+        rm -f "${old_dep_dep_on_file}"
+    fi
+
+    # Remove the deps of the old dep recursively, of course only, when it does not have any other dependants
+    local old_dep_dep_file
+    old_dep_dep_file="${TUE_INSTALL_DEPENDENCIES_DIR}"/"${old_dep_target}"
+    if [[ -f "${old_dep_dep_file}" ]]
+    then
+        # Iterate over all depencies of old_dep_target, which is removed.
+        while read -r dep_of_old_dep
+        do
+            # Actually remove the deps
+            local dep_error
+            _remove_old_target_dep_recursively "${old_dep_target}" "${dep_of_old_dep}"
+            dep_error=$?
+            if [ ${dep_error} -gt 0 ]
+            then
+                error_code=1
+            fi
+
+        done < "${old_dep_dep_file}"
+        rm -f "${old_dep_dep_file}"
+    else
+        tue-install-debug "[remove_old_dep] No depencies file exist for target: ${old_dep_target}"
+    fi
+
+    tue-install-debug "[remove_old_dep] Fully uninstalled '${old_dep_target}' as a dependency of '${parent_target}'"
```

Comment:
Convert into `tue-install-info` So people know they can remove this pkg from their workspace